packages/block-editor/src/components/block-list/hover-area.js

```diff
@@ -1,82 +1,43 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { withSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
 
-class HoverArea extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			hoverArea: null,
-		};
-		this.onMouseLeave = this.onMouseLeave.bind( this );
-		this.onMouseMove = this.onMouseMove.bind( this );
-	}
-
-	componentWillUnmount() {
-		if ( this.props.container ) {
-			this.toggleListeners( this.props.container, false );
-		}
-	}
-
-	componentDidMount() {
-		if ( this.props.container ) {
-			this.toggleListeners( this.props.container );
-		}
-	}
-
-	componentDidUpdate( prevProps ) {
-		if ( prevProps.container === this.props.container ) {
-			return;
-		}
-		if ( prevProps.container ) {
-			this.toggleListeners( prevProps.container, false );
-		}
-		if ( this.props.container ) {
-			this.toggleListeners( this.props.container, true );
-		}
-	}
+const useHoveredArea = ( wrapper ) => {
+	const [ hoveredArea, setHoveredArea ] = useState( null );
 
-	toggleListeners( container, shouldListnerToEvents = true ) {
-		const method = shouldListnerToEvents ? 'addEventListener' : 'removeEventListener';
-		container[ method ]( 'mousemove', this.onMouseMove );
-		container[ method ]( 'mouseleave', this.onMouseLeave );
-	}
-
-	onMouseLeave() {
-		if ( this.state.hoverArea ) {
-			this.setState( { hoverArea: null } );
-		}
-	}
+	useEffect( () => {
+		const onMouseLeave = () => {
+			if ( hoveredArea ) {
+				setHoveredArea( null );
+			}
+		};
 
-	onMouseMove( event ) {
-		const { isRTL, container } = this.props;
-		const { width, left, right } = container.getBoundingClientRect();
+		const onMouseMove = ( event ) => {
+			const { width, left, right } = wrapper.current.getBoundingClientRect();
 
-		let hoverArea = null;
-		if ( ( event.clientX - left ) < width / 3 ) {
-			hoverArea = isRTL ? 'right' : 'left';
-		} else if ( ( right - event.clientX ) < width / 3 ) {
-			hoverArea = isRTL ? 'left' : 'right';
-		}
+			let newHoveredArea = null;
+			if ( ( event.clientX - left ) < width / 3 ) {
+				newHoveredArea = 'left';
+			} else if ( ( right - event.clientX ) < width / 3 ) {
+				newHoveredArea = 'right';
+			}
 
-		if ( hoverArea !== this.state.hoverArea ) {
-			this.setState( { hoverArea } );
-		}
-	}
+			if ( newHoveredArea !== hoveredArea ) {
+				setHoveredArea( newHoveredArea );
+			}
+		};
 
-	render() {
-		const { hoverArea } = this.state;
-		const { children } = this.props;
+		wrapper.current.addEventListener( 'mousemove', onMouseMove );
+		wrapper.current.addEventListener( 'mouseleave', onMouseLeave );
 
-		return children( { hoverArea } );
-	}
-}
+		return () => {
+			wrapper.current.removeEventListener( 'mousemove', onMouseMove );
+			wrapper.current.removeEventListener( 'mouseleave', onMouseLeave );
+		};
+	}, [] );
```

Comment:
could `wrapper.current` be the dependency here?

Reply:
`wrapper` is a mutable object (like all refs), the reasoning here is that this hook is only supported from DOM elements which means the wrapper don't change at all, which also means, there's no need for a dependency here. 

I checked similar third-party hooks and they seem to share my reasoning 
https://github.com/FezVrasta/react-resize-aware/blob/master/src/useOnResize.js#L26

https://github.com/SoftBind/react-hooks/blob/master/packages/useMeasure/src/useMeasure.js#L30